source/adios2/toolkit/transport/file/FilePOSIX.cpp

```diff
@@ -59,7 +59,9 @@ void FilePOSIX::Open(const std::string &name, const Mode openMode,
 {
     auto lf_AsyncOpenWrite = [&](const std::string &name) -> int {
         ProfilerStart("open");
+        errno = 0;
```

Comment:
I'm confused about this. You set `errno` to zero, then assign `m_Errno` to `errno`, without any intervening assignments. So `m_Errno` is always zero.

Reply:
Oh wow I just read about `errno`; I guess it behaves like an `eflags` register.

Reply:
There is an intervening system function call that sets 'errno' to non-zero if there is an error. It's standard practice (for C programmers at least) to zero out errno before system calls to not catch errors from the past, hence the errno=0 statement here. m_Errno is used to remember the (possible) error for the call right after this and checked later (possibly the call in happens in a thread, line 63, but the check is in main thread much later)